### PR TITLE
Add new package: oozie

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

**Review by alalazo — CHANGES_REQUESTED** [review]
I think there might be dependencies missing

File: var/spack/repos/builtin/packages/oozie/package.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Oozie(MavenPackage):
+    """Oozie is an extensible, scalable and reliable system to define,
+    manage, schedule, and execute complex Hadoop workloads via web services."""
+
+    homepage = "https://oozie.apache.org/"
+    url      = "https://github.com/apache/oozie/archive/release-5.2.0.tar.gz"
+
+    version('5.2.0', sha256='998741fb64b0badef917586685f8da90c5f037da9ed187361ac8db888780bf14')
+
+    depends_on('java@8', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/oozie/package.py:18]: Also add:
```python
depends_on('maven@3:', type='build')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/oozie/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/oozie/package.py:18]: Fixed this in https://github.com/spack/spack/pull/18550/commits/896d25ba7322122c5678b8d786ae4f011f873692

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@alalazo can you elaborate? Most Maven packages in Spack right now have hundreds of dependencies that aren't packaged. I agree, we should add packages for them instead of allowing the build system to download them automatically, but I don't have the time or knowledge to enforce this or add the packages myself.

**Comment by darmac** [comment]
@alalazo  Is there any more dependencies I need add to this package?

**Comment by alalazo** [comment]
@darmac @adamjstewart 

> Most Maven packages in Spack right now have hundreds of dependencies that aren't packaged. I agree, we should add packages for them instead of allowing the build system to download them automatically, but I don't have the time or knowledge to enforce this or add the packages myself.

I was not aware of that. Looking at [System Requirements](http://oozie.apache.org/docs/5.0.0/DG_QuickStart.html) in the docs page, it seems that `oozie` depends on `hadoop` and `pig`. Should we add these dependencies to the package? If not, I'm fine merging this package as is for now.

**Comment by darmac** [comment]
Yes, it needs `hadoop` and `pig`, but `hadoop` is unavailable for `aarch64` in spack now.
Maybe I should fix `hadoop` before merge that package.

PS: hadoop patch - https://github.com/spack/spack/pull/17017

**Pull request closed without merging**